src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -482,6 +514,30 @@ class GroupReadsByUmi
     }
   }
 
+  /** When a minimum UMI length is specified, truncates all the UMIs to the length of the shortest UMI.  For the paired
+    * assigner, truncates the first UMI and second UMI sepeartely.*/
+  private def truncateUmis(umis: Seq[Umi]): Seq[Umi] = this.minUmiLength match {
+    case None => umis
+    case Some(_minUmiLength) =>
+      this.assigner match {
+        case _: PairedUmiAssigner => Seq.empty
```

Comment:
I'm not even sure we should support this for `paired`. It's so much more complicated, and even now this doesn't (I think) guarantee that the resulting pairs have len > min length.  E.g. if min length is 8 and you get `ACGTA-ACG` and `ACG-ACGGG` I think you'll reduce both to `ACG-ACG`.  Also, it's necessary that the left and right UMIs be the same length for duplex to work.

Reply:
Yup I did the best I could.

Reply:
@tfenne I am torn as well, I'll update to remove support in the pairedumiassigner, which I think will simplify things a bit.